src/Grpc.AspNetCore.Server/InterceptorCollection.cs

```diff
@@ -0,0 +1,97 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using Grpc.Core.Interceptors;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Grpc.AspNetCore.Server
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class InterceptorCollection : IEnumerable<InterceptorRegistration>
+    {
+        private static readonly IEnumerator<InterceptorRegistration> EmptyEnumerator = Enumerable.Empty<InterceptorRegistration>().GetEnumerator();
+
+        private List<InterceptorRegistration> _store;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        public bool IsEmpty => _store == null;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        public IReadOnlyList<InterceptorRegistration> Snapshot => _store?.AsReadOnly(); 
```

Comment:
What is this for? At the very least this should be a method instead of a property.